helion/_compiler/indexing_strategy.py

```diff
@@ -447,6 +450,14 @@ def codegen_store(
         )
 
 
+def _calculate_ellipsis_dims(
+    index: list[object], current_index: int, total_dims: int
+) -> int:
+    """Calculate how many dimensions an ellipsis should expand to."""
+    remaining_indices = len(index) - current_index - 1
+    return total_dims - current_index - remaining_indices
```

Comment:
I don't think this is right when combined with `None` (which will increment `i` in the caller but not consume an input dimension.